Sources/NostrSDK/DirectMessageEncrypting.swift

```diff
@@ -0,0 +1,170 @@
+//
+//  DirectMessageEncrypting.swift
+//  
+//
+//  Created by Joel Klabo on 8/10/23.
+//
+
+import Foundation
+import secp256k1
+import CommonCrypto
+import CryptoKit
+
+public enum DirectMessageEncryptingError: Error {
+    case pubkeyParsing
+    case sharedSecretCreation
+    case encryptionError
+    case decryptionError
+    case missingValue
+}
+
+public protocol DirectMessageEncrypting {}
+public extension DirectMessageEncrypting {
+
+    func encrypt(content: String, privateKey: PrivateKey, publicKey: PublicKey) throws -> String {
+
+        guard let sharedSecret = try? getSharedSecret(privateKey: privateKey, recipient: publicKey) else {
+            throw EventCreatingError.invalidInput
```

Comment:
Since this function throws, why ignore the failed call and throw a custom error? You could just call:
```swift
let sharedSecret = try getSharedSecret(...
```